From cff938227d787a9f5f97fd74bc456a3315e66166 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:55 -0400
Subject: [usb] usbtest: Always clear halt else further tests will fail

Message-id: <1410887463-20674-224-git-send-email-dzickus@redhat.com>
Patchwork-id: 95672
O-Subject: [RHEL7 PATCH 223/591] usb: usbtest: Always clear halt else further tests will fail
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 824d752b04765fc513fe17666a539f6c73960c4e
Author: Roger Quadros <rogerq@ti.com>
Date:   Wed Dec 18 15:40:11 2013 +0530

    usb: usbtest: Always clear halt else further tests will fail

    In test_halt() we set an endpoint halt condition and return on halt verification
    failure, then the enpoint will remain halted and all further tests related
    to that enpoint will fail. This is because we don't tackle endpoint halt error condition
    in any of the tests. To avoid that situation, make sure to clear the
    halt condition before exiting test_halt().

    Signed-off-by: Roger Quadros <rogerq@ti.com>
    Reviewed-by: Felipe Balbi <balbi@ti.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/misc/usbtest.c b/drivers/usb/misc/usbtest.c
index a782bfc..f6568b5 100644
--- a/drivers/usb/misc/usbtest.c
+++ b/drivers/usb/misc/usbtest.c
@@ -1554,8 +1554,17 @@ static int test_halt(struct usbtest_dev *tdev, int ep, struct urb *urb)
   return retval;
  }
  retval = verify_halted(tdev, ep, urb);
- if (retval < 0)
+ if (retval < 0) {
+  int ret;
+
+  /* clear halt anyways, else further tests will fail */
+  ret = usb_clear_halt(urb->dev, urb->pipe);
+  if (ret)
+   ERROR(tdev, "ep %02x couldn't clear halt, %d\n",
+         ep, ret);
+
   return retval;
+ }
 
  /* clear halt (tests API + protocol), verify it worked */
  retval = usb_clear_halt(urb->dev, urb->pipe);
-- 
1.7.1